app/main/posts/modify/post-data-editor.directive.js

```diff
@@ -89,6 +89,10 @@ function PostDataEditorController(
     $scope.selectForm = selectForm;
     $scope.isSaving = LoadingProgress.getSavingState;
 
+    $scope.cancel = function () {
+        $location.path('/views/data');
```

Comment:
We want to keep the post selected here, instead of going to the general data-view. Hint: if you look at this row (its the function for the upper cancel-button in the "toolbar"): https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/views/post-toolbar.directive.js#L53, it goes to the data-view with the post still selected.